packages/react-components/src/icons/src/Icon.tsx

```diff
@@ -60,27 +73,31 @@ export function InnerIcon(props) {
             )}
         />
     );
-}
+}) as InnerIconComponentWithFactory;
 
-InnerIcon.propTypes = propTypes;
+type IconComponentWithFactory = OrbitComponent<"svg", InnerIconProps> & {
+    create: IconFactory
+}
 
-export const Icon = slot("icon", forwardRef((props, ref) => (
+export const Icon = slot("icon", forwardRef<InnerIconProps, "svg">((props, ref) => (
     <InnerIcon {...props} forwardedRef={ref} />)
-));
+)) as IconComponentWithFactory;
 
 Icon.displayName = "Icon";
 
+export type IconProps = ComponentProps<typeof Icon>;
+
 ////////
 
-function createIconFactory(type) {
-    return slot("icon", forwardRef((props, ref) =>
+const createIconFactory: IconFactory = type => {
```

Comment:
just wondering, is it a must to convert the function declaration to an arrow function here to use TS?